logstash-core/lib/logstash/pipeline.rb

```diff
@@ -696,10 +701,21 @@ def stalling_threads_info
       .each {|t| t.delete("status") }
   end
 
+  def collect_dlq_stats
+    dlq_metric = @metric.namespace([:stats, :pipelines, pipeline_id.to_s.to_sym, :dlq])
+    enabled = settings.get_value("dead_letter_queue.enable")
+    dlq_metric.gauge(:enabled, enabled)
+    if enabled
+      dlq_metric.gauge(:path, @dlq_writer.get_path.to_absolute_path)
+      dlq_metric.gauge(:segment_count, @dlq_writer.get_segment_count)
```

Comment:
Possibly not... I found it useful, as I wasn't aware that a new segment was created every time logstash starts up before I saw the stat ;)

I'm happy to remove it, if it is superfluous, though